### ISSUE TITLE
(in continuation of discussion in ec-/Quake3e/commit/50167f34e361bac156315e53efccb5a5d4acac80)

### ISSUE THREAD

**Issue by makise-homura** [makise-homura/empty#1]
A reply to @ec- for https://github.com/ec-/Quake3e/commit/50167f34e361bac156315e53efccb5a5d4acac80

> What about this person https://sdelanounas.ru/blogs/132891

He's not my close friend, and my and his positions on this subject are completely opposite. I do not support him on his opinion and AFAIK, he himself didn't do anything on spoken commits and anything else of subject discussed here.

I didn't state EVERYONE is against the war, I state that MOST OF US are.

> key russian opensource sites

I don't use such sites and don't want to break FOSS community to Russian and non-Russian parts. That's why I'm here on GitHub, not on such sites you talk about (I actually don't know which ones do you mean).

> including ones related to Elbrus development/support

Don't lie again. Developing ALT Linux has nothing to do with developing Elbrus CPU.

> come here again after building civil society that cares about its rights

Come on, let me know how to do that. If you asking ME to do that, I ask YOU to start nuclear bombardment of Moscow. _You can do it if you want_, I guess. Everyone will get what they want from that: you will have me killed, and I won't bother you anymore by such discussions, and international society will have Russia devastated and not being able to plant wars anymore. Come on, I wait for your actions.

> Until that - it is your fault that you (silently and continuously) allow your president/government start and continue this war.

Care to explain what I did to fail? My own actions, not government ones?

> And I'm pretty sure you know what to do on some Square to stop that world-wide 'rusophobia' stream in short term

What? I really don't know. Start a protest? It doesn't work, many people tried since 2011. Get a weapon and kill Putin? Until you send this weapon and coordinates of Putin, to someone who's bold enough, it's impossible too. We have no weapons on hand, contrary to how it's in Ukraine now. Any other ideas?

> Even after Crimea annexion I was tolerant enough to accept initial Elbrus support but first cruise missile strikes that hit my peaceful city

Neither authors of these commits, nor me has any involvement in these strikes.

> I will reject everything that has even tiny relation to current russian state/government/military

Care to explain how Quake has a relation to state/government/military?

> as the rest of the world does.

You're misinformed. Even GitHub states that FOSS infrastructure and stuff must be _available to all, including developers in Russia_.

> So either do the right thing on mentioned square

Come on, say what I should do then. Red Square is not Maidan, after all. It's not quite possible to perform a revolution by visiting it.

tl;dr: Just imagine yourself at my place. Like, you feel totally incapable of doing anything with that nonsense your goverment somehow started, you're just sitting reading news and watching how your life gets more and more screwed up. And, on the other side, people who have no idea how do you feel about that, accuse you of the things you never ever accept in your life, and try to _cancel_ all contributions you and your collaborators done out of the sheer entusiasm, and declare this enthusiasm a support of such cringe actions.

**Comment by makise-homura** [comment]
Btw, an original message (here it'll be safe):

So that's your precious free speech, yes? When @ec- publish a clear lie, like in second message in this thread, but it's pro-Ukrainian, it's ok; when I write how it is in reality (because I work in INEUM, close companion of MCST and know how it is from inside), but my words protect Russian developers (not government! I especially state: I'm against our government's actions!), it's something you should delete, right?

Please note: I highly oppose decision of Putin to start a war and consider in one of most stupid decisions of him. But I guess that doesn't matter to you, your actions just make an evidence that you don't hate Putin, you hate all Russians, especially enthusiasts. Please read the [GitHub point](https://github.blog/2022-03-02-our-response-to-the-war-in-ukraine/) on that subject, and follow it. 

@ec- 

> I found disappointing finding no evidence they're doing anything against their government

BECAUSE WE CAN'T. If you have any idea how to do this, I highly appreciate you tell us how, but I guess you haven't. Protests are not working in Russia and Belarus at all, it is clearly confirmed (but even still, there are everyday anti-war protests out of desperation in Moscow and St.Petersburg, the only result of which is more and more people arrested). No one of protesters in Russia have any success since 2011, and even latest massive protests in Belarus had no effect. People has no firearms, and thus have absolutely no power to commit a revolution or whatever. The only way to destroy Putin regime is a military invasion of NATO, massive nuclear attack on major Russian cities and infrastructure, and a wide massacre of Russian people (because military will hide beneath them). But neither you, neither I don't want it. At least I hope for it.

> were happy after russia annexed Crimea in 2014

You have no idea about it. Probably you just listened to propaganda and people that follows it. Annex of Crimea was a cause of major downfall of quality of life in Russia since 2014 (incomparable to today's, but still a thing). I have nothing went better in my life because of Crimea, and neither had any of my close friends and people who actually have a mind to think, not just follow propaganda. So people who have a mind, don't approve annex of Crimea and support of Donbass separatists.

> unfair financial sanctions to peaceful russian people

Yes, because current sanctions really target people who has nothing to do with the war, who mostly don't support it, who lost many things because of the war, and who will lost even more because they can't buy anything and lost their work because of major economy fault. Government and military have no problem to rob people of their money (they did it already in 1991, 1993, 1998, 2008, and 2019) for the war, so USA/Europe won't stop the war because of sanctions; but people in Russia will suffer more and more. Europe and USA think that "if we turn people into beggars, they will gather together and make a revolution", but no, they are wrong. They never lived in authoritarian state and can't imagine it won't work--but it won't work. We tried to throw out Putin since 2011, and there was no success at all, and sanctions won't do any better.

> 'russophoby' is a known and well-used term of russian propaganda

Then why are you confirming that it is not just a propaganda term, but really existing thing, by reverting commits basing on nationality of those who made these commits and made CPU that is supported by these commits? (arguments aginst military is really obscure, because military don't use ioquake3 and Quake3e, and barely use Elbrus CPUs). Ok, let's use a different term, like 'offending contributors somehow connected to Russia'. Be open for anyone all over the world, be it a contributor from USA or Europe, or Russia or North Korea. Then, and only then, _offense to contributors somehow connected to Russia_ will remain a propaganda term. I feel _offense to contributors somehow connected to Russia_ when people write things like "kill all Russians" on Twitter and Facebook, when they blame my friend in Germany that he "started the war" when he speaks Russian in public places with his companions (how can he start the war whe he's out of Russia since 2017?), and when someone reverts commits that me and my other friends made by attributing it to "Russian military pigs". I'm not a military, and not a pig, and neither any of us in E2K community, but you call us that. What is it if not _offense to contributors somehow connected to Russia_ then?


P.S. By the way, your efforts on banning free speech due to pro-Ukrainian position are in vain: I have a keylogger and easily can restore what I once wrote, because everything is saved:

@ec- Things you said has nothing to do with reality. MCST is not so tightly related with MoD, even more, MoD uses mostly SPARC chips, not E2K. E2K is used now primarily by business (both government and commercial), not by defense. You can buy Elbrus by contacting any dealer (e.g. Bitblaze, Imaxai, etc.: MCST itself does not sell platforms to end users, as any other major supplier does; this is the subject of dealers), or even get a SSH access free and without any obligations (by contacting me). It's not a "closed military platform" today, as some people pretend.

@NuclearMonster I consider patch removal that is based on _offense to contributors somehow connected to Russia_ and other ideological purposes a completely inappropriate in FOSS philiosophy. Especially in the case it does not obviously help any real problem to be solved (you really think if you remove e2k support from ioquake and ban everyone who object this, it will make Russian forces to feel offended and retreat from Ukraine? Think again then.)

tl;dr: by reverting these patches, you hurt community (who's mostly against the war, not Russian military), so your actions actually causing the reverse of what you wanted.

@icculus I really don't get your reason for discriminating contributors from Russia. Please note that people in Russia has absolutely no control over its government (it's not a democracy, it's authoritarian state), and there is no reason to think like "every contributor is equal, but those not from Russia are more equal than Russian", because contributions made by Russian people have nothing in common with Putin's actions and Russia's foreign policy.


Due to informal campaign targeted against Russian-based developers and their contributions (which is considered inappropriate by [GitHub](https://github.blog/2022-03-02-our-response-to-the-war-in-ukraine/)), user @ec- acts the way I consider offensive for me and other Russian developers.
* They publishes lies and outdated information that offends Elbrus community and people who contributes to it in [this](https://github.com/ec-/Quake3e/commit/50167f34e361bac156315e53efccb5a5d4acac80#commitcomment-67518912) and [this](https://github.com/ec-/Quake3e/commit/50167f34e361bac156315e53efccb5a5d4acac80#commitcomment-68569131) comments (If you want proof about why most of their words is a lie, I have, and can provide such proofs);
* They calls contributors who made a commit they reverted "russian military pigs" (according to name of commit), even if they clearly state that they don't support Russia's aggression;
* And they accepts no explanation from other side of conflict, deleting their comments and silently banning them without any reason (I have screenshots of my messages that were deleted from this thread, and can provide them too).

**Comment by icculus** [comment]
> @icculus I really don't get your reason for discriminating contributors from Russia.

I don't control either this project or ioquake3; I was just apologizing for merging a commit that made the project owners upset.

It seems to me that you and ec- are both in an awful situation that neither of you wanted nor are at fault for, and I think you are both right to be angry. I didn't mean to make it worse for either of you; I just merged a PR because someone emailed to ask me to review it.

I hope you will both accept my apology for unintentionally causing this fight.

**Comment by makise-homura** [comment]
@icculus Yeah, you're completely right: none of us wanted this (at least I hope so), but the situation placed us on different sides of barricades, so it's unfortunately gone like that.

As you explain, I see you address only a regret of the fact that this commit made someone upset, not of the fact this commit was merged at all; if it doesn't lead to a decision of avoiding merge of any commit from Russian developers, especially Elbrus community, then it's no problem with that. My apologies for getting you incorrectly then.

The thing I try to achieve, is to avoid prejudice, especially the one based on propaganda, and to stop the real-life government conflicts being extended into FOSS community. I always wanted to see developers as people, between whom there are no barriers of nationality, race, social group, religion and any other real-life stuff, and where people always can find a solution to any problem through constructive conversation, rather than fighting each other, banning people with opposite point of view and deleting comments. And when someone is starting to hate Russian Elbrus community because of unjust accusations, when they revert commits that actually don't harm anyone, explaining this by ideology and calling commit authors "military pigs", and last but not least--when they deny any right of other side of conflict to answer in defense, it barely can be called constructive.

I hope you judge and will judge any commit and PR suggested, by its use for the project, not by who made it and which community they participate in. If I'm right, there's not a single complaint for you.

**Comment by a1batross** [comment]
@icculus thanks for looking at this adequately. 

I initially asked you (the original post was deleted) because @NuclearMonster is just unreachable. Anyone who raises this discussion with them seems to get instantly banned. I'm not able to post at @ioquake anymore and other user reported that he got banned from their Discord and Twitter, although I can understand this, it's their personal space.

While I can understand this kind of behavior from ec-, because yes, they as a Ukrainian has all right to be mad. But I don't get how NuclearMonster got involved in this, and this feels wrong.

Especially if we look at the reverted changes in both ec-'s project and ioquake3, not they're just zero effort to maintain, but they're also useless. Yes, in fact, one can still build the game from source. It's nothing but pure emotions.
